cli/src/commands/runTests.js

```diff
@@ -578,34 +616,101 @@ async function runTestGroup(
+          depPaths,
+        );
+        const flowErrorsForThisGroup = await runFlowTypeDefTests(
+          [...sublistOfFlowVersions],
+          testGroup.id,
+          testDirPath,
+        );
+        flowErrors.push(...flowErrorsForThisGroup);
+      }
+
+      const lowestFlowVersionRan = flowVersionsToRun[0];
+      const lowestCapableFlowVersion = await findLowestCapableFlowVersion(
         repoDirPath,
+        orderedFlowVersions,
+        lowestFlowVersionRan,
         testDirPath,
         testGroup.libDefPath,
-        lowestFlowVersionRanInThisGroup,
-      );
-      const flowErrorsForThisGroup = await runFlowTypeDefTests(
-        sublistOfFlowVersions,
-        testGroup.id,
-        testDirPath,
+        depPaths,
       );
-      flowErrors.push(...flowErrorsForThisGroup);
-    }
-
-    const lowestFlowVersionRan = flowVersionsToRun[0];
-    const lowestCapableFlowVersion = await findLowestCapableFlowVersion(
-      repoDirPath,
-      orderedFlowVersions,
-      lowestFlowVersionRan,
-      testDirPath,
-      testGroup.libDefPath,
-    );
 
-    if (lowestCapableFlowVersion !== lowestFlowVersionRan) {
-      console.log(`Tests for ${testGroup.id} ran successfully on flow ${lowestCapableFlowVersion}.
+      if (lowestCapableFlowVersion !== lowestFlowVersionRan) {
+        console.log(`Tests for ${testGroup.id} ran successfully on flow ${lowestCapableFlowVersion}.
         Consider setting ${lowestCapableFlowVersion} as the lower bound!`);
+      }
+    };
+
+    if (Object.keys(testGroup.deps).length > 0) {
+      // Map through every dependency and their versions and replace with
+      // the definition's path.
+      // Then shuffle to create a new Array<Array<>> that will test
+      // All dependencies across various supported versions.
+      const depsTestGroups = (() => {
+        const flowDirVersion = extractFlowDirFromFlowDirPath(testGroup.id);
+        const depBasePath = getNpmLibDefDirFromNested(testGroup.libDefPath);
+
+        const mappedDepPaths = Object.keys(testGroup.deps).map(depName => {
+          const nameSplit = depName.split('/');
+          const scope = nameSplit.length > 1 ? `${nameSplit[0]}/` : '';
+          const packageName = nameSplit.length > 1 ? nameSplit[1] : depName;
+
+          return testGroup.deps[depName].map(version => {
+            const tempPath = `${depBasePath}${scope}deps_${packageName}_${version}/${flowDirVersion}/${packageName}_${version}.js`;
+            const path = `${depBasePath}${scope}${packageName}_${version}/${flowDirVersion}/${packageName}_${version}.js`;
+            if (fs.existsSync(tempPath)) {
+              return tempPath;
+            }
+
+            if (!fs.existsSync(path)) {
+              throw new Error(
+                colors.red(
+                  `${depName}@${version} cannot be a dependency of ${testGroup.id} because either the dependency@version does't exist or they do not have matching flow version ranges`,
```

Comment:
Maybe link to the relevant bit in README to guide the person to resolution. I'm imagining a contributor bumping into this error and being unclear on how to resolve it. It'd be great to give some suggestions in the docs.